frontend/public/components/cluster-service-class-icon.tsx

```diff
@@ -6,13 +6,105 @@ import * as classNames from 'classnames';
+  'icon-decisionserver': 'decisionserver.svg',
+  'icon-django': 'django.svg',
+  'icon-dotnet': 'dotnet.svg',
+  'icon-drupal': 'drupal.svg',
+  'icon-eap': 'eap.svg',
+  'icon-elastic': 'elastic.svg',
+  'icon-erlang': 'erlang.svg',
+  'icon-fedora': 'fedora.svg',
+  'icon-freebsd': 'freebsd.svg',
+  'icon-git': 'git.svg',
+  'icon-github': 'github.svg',
+  'icon-gitlab': 'gitlab.svg',
+  'icon-glassfish': 'glassfish.svg',
+  'icon-go-gopher': 'go-gopher.svg',
+  'icon-grails': 'grails.svg',
+  'icon-hadoop': 'hadoop.svg',
+  'icon-haproxy': 'haproxy.svg',
+  'icon-infinispan': 'infinispan.svg',
+  'icon-jboss': 'jboss.svg',
+  'icon-jenkins': 'jenkins.svg',
+  'icon-jetty': 'jetty.svg',
+  'icon-joomla': 'joomla.svg',
+  'icon-jruby': 'jruby.svg',
+  'icon-js': 'js.svg',
+  'icon-kubevirt': 'kubevirt.svg',
+  'icon-laravel': 'laravel.svg',
+  'icon-load-balancer': 'load-balancer.svg',
+  'icon-mariadb': 'mariadb.svg',
+  'icon-mediawiki': 'mediawiki.svg',
+  'icon-memcached': 'memcached.svg',
+  'icon-mongodb': 'mongodb.svg',
+  'icon-mssql': 'mssql.svg',
+  'icon-mysql-database': 'mysql-database.svg',
+  'icon-nginx': 'nginx.svg',
+  'icon-nodejs': 'nodejs.svg',
+  'icon-openjdk': 'openjdk.svg',
+  'icon-openshift': 'openshift.svg',
+  'icon-openstack': 'openstack.svg',
+  'icon-other-linux': 'other-linux.svg',
+  'icon-other-unknown': 'other-unknown.svg',
+  'icon-perl': 'perl.svg',
+  'icon-phalcon': 'phalcon.svg',
+  'icon-php': 'php.svg',
+  'icon-play': 'play.svg',
+  'icon-postgresql': 'postgresql.svg',
+  'icon-processserver': 'processserver.svg',
+  'icon-python': 'python.svg',
+  'icon-rabbitmq': 'rabbitmq.svg',
+  'icon-rails': 'rails.svg',
+  'icon-redis': 'redis.svg',
+  'icon-rh-integration': 'rh-integration.svg',
+  // Use the upstream icon.
+  'icon-rh-openjdk': 'openjdk.svg',
+  'icon-rh-tomcat': 'rh-tomcat.svg',
+  'icon-ruby': 'ruby.svg',
+  'icon-scala': 'scala.svg',
+  'icon-shadowman': 'shadowman.svg',
+  'icon-spring': 'spring.svg',
+  'icon-sso': 'sso.svg',
+  'icon-stackoverflow': 'stackoverflow.svg',
+  'icon-suse': 'suse.svg',
+  'icon-symfony': 'symfony.svg',
+  'icon-tomcat': 'tomcat.svg',
+  'icon-ubuntu': 'ubuntu.svg',
+  'icon-wildfly': 'wildfly.svg',
+  'icon-windows': 'windows.svg',
+  'icon-wordpress': 'wordpress.svg',
+  'icon-xamarin': 'xamarin.svg',
+  'icon-zend': 'zend.svg'
+});
+
 const normalizeIconClass = (iconClass) => {
   return _.startsWith(iconClass, 'icon-') ? `font-icon ${iconClass}` : iconClass;
 };
 
 export const ClusterServiceClassIcon: React.SFC<ClusterServiceClassIconProps> = ({serviceClass, iconSize}) => {
-  const imageUrl = _.get(serviceClass, ['spec', 'externalMetadata', 'imageUrl']);
   const iconClass = _.get(serviceClass, ['spec', 'externalMetadata', 'console.openshift.io/iconClass'], 'fa fa-clone');
+  const iconClassImg = LOGOS[iconClass];
+  const imageUrl = _.get(serviceClass, ['spec', 'externalMetadata', 'imageUrl']) || iconClassImg && `static/assets/${iconClassImg}`;
```

Comment:
[One of the major points of ES6 modules is to avoid magic URL strings in favor of statically-defined imports for your entire app](https://stackoverflow.com/a/42118921) (you should _never_ reference the build directory, only source files). When you `import` an image, the  [`file-loader`](https://github.com/webpack-contrib/file-loader) will include it in the bundle and assign the variable the unique URL of the file. See how we [use images in the sidenav](https://github.com/openshift/console/blob/master/frontend/public/components/nav.jsx#L244) for an example.

Remove the `require.context` and use:
```typescript
import * as apacheImg from '../imgs/logos/apache.svg';
// All the rest

const logos = new Map()
  .set('icon-apache', apacheImg)
  // All the rest

...

const imageUrl = _.get(serviceClass, ['spec', 'externalMetadata', 'imageUrl']) || iconClassImg && logos.get(iconClassName);
```